src/disjoint_mut.rs

```diff
@@ -85,6 +106,23 @@ pub struct DisjointImmutGuard<'a, T: AsMutPtr, V: ?Sized> {
     bounds: Bounds,
 }
 
+impl<'a, T: AsMutPtr> DisjointImmutGuard<'a, T, [u8]> {
```

Comment:
It's fine for now, but as a future potential optimization, we should see if we can add alignment info to the `[u8]` here, as without it, `fn cast` has to do an alignment check, even though the buffer we're indexing into is usually already aligned enough.

Reply:
Good point.